File: dotnet/CardsProject/Domain/Marketplace/TradeDispute.cs
using CardsProject.Domain.Players;
using System.ComponentModel.DataAnnotations.Schema;

namespace CardsProject.Domain.Marketplace;

public enum TradeDisputeStatusType
{
    Open,
    UnderReview,
    Resolved,
    Escalated
}

public enum TradeDisputeReasonType
{
    ItemNotReceived,
    ItemNotAsDescribed,
    FraudSuspected,
    Other
}

public class TradeDispute
{
    public int Id { get; set; }

    public TradeDisputeStatusType Status { get; set; }
    public TradeDisputeReasonType Reason { get; set; }
    public string Description { get; set; } = "";
    public string? Resolution { get; set; }
    public DateTime? OpenedAt { get; set; } = null;
    public DateTime? ResolvedAt { get; set; } = null;

    public int? TransactionId { get; set; }
    [ForeignKey(nameof(TransactionId))]
    public TradeTransaction? Transaction { get; set; }
    public int? OpenedById { get; set; }
    [ForeignKey(nameof(OpenedById))]
    public Player? OpenedBy { get; set; }
    public int? ResolvedById { get; set; }
    [ForeignKey(nameof(ResolvedById))]
    public Player? ResolvedBy { get; set; }

    // Business operations

    public void Escalate()
    {
        // TODO: implement escalate
    }

    public void Resolve(string resolutionText)
    {
        // TODO: implement resolve
    }

    public void CloseResolved()
    {
        // TODO: implement close_resolved
    }

    public void Review()
    {
        // TODO: implement review
    }

    // ── Lifecycle state machine ──────────────────────────────────────
    private static readonly System.Collections.Generic.Dictionary<TradeDisputeStatusType, TradeDisputeStatusType[]> AllowedTransitions = new()
    {
        [TradeDisputeStatusType.Open] = new[] { TradeDisputeStatusType.UnderReview },
        [TradeDisputeStatusType.UnderReview] = new[] { TradeDisputeStatusType.Resolved, TradeDisputeStatusType.Escalated },
        [TradeDisputeStatusType.Escalated] = new[] { TradeDisputeStatusType.Resolved }
    };

    public void AssertTransition(TradeDisputeStatusType to)
    {
        if (!AllowedTransitions.TryGetValue(Status, out var allowed) || !System.Array.Exists(allowed, s => s == to))
            throw new InvalidOperationException($"Transition {Status} -> {to} not allowed");
    }
}

File: laravel/app/Models/Marketplace/TradeDispute.php
<?php

namespace App\Models\Marketplace;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use App\Models\Players\Player;

class TradeDispute extends Model
{
    protected $table = 'trade_disputes';

    protected $fillable = ['status', 'reason', 'description', 'resolution', 'opened_at', 'resolved_at', 'transaction_id', 'opened_by_id', 'resolved_by_id'];

    protected $casts = [
        'opened_at' => 'datetime',
        'resolved_at' => 'datetime',
    ];

    const STATUS_VALUES = ['Open', 'UnderReview', 'Resolved', 'Escalated'];
    const REASON_VALUES = ['ItemNotReceived', 'ItemNotAsDescribed', 'FraudSuspected', 'Other'];

    public function transaction(): BelongsTo
    {
        return $this->belongsTo(TradeTransaction::class, 'transaction_id');
    }

    public function openedBy(): BelongsTo
    {
        return $this->belongsTo(Player::class, 'opened_by_id');
    }

    public function resolvedBy(): BelongsTo
    {
        return $this->belongsTo(Player::class, 'resolved_by_id');
    }

    // ── Domain invariants (IMPLIES rules) ───────────────────────────────
    public function validateImplies(): void
    {
        if ($this->resolved_at !== null && !($this->status === 'Resolved')) {
            throw new \RuntimeException('resolved_at_requires_terminal_status');
        }
    }

    // ── Lifecycle state machine ──────────────────────────────────────
    const ALLOWED_TRANSITIONS = [
        'Open' => ['UnderReview'],
        'UnderReview' => ['Resolved', 'Escalated'],
        'Escalated' => ['Resolved'],
    ];

    public function assertTransition(string $to): void
    {
        $allowed = static::ALLOWED_TRANSITIONS[$this->status] ?? [];
        if (!in_array($to, $allowed, true)) {
            throw new \InvalidArgumentException("Transition {$this->status} -> {$to} not allowed");
        }
    }

    // ── Business operations ──────────────────────────────────────────

    public function escalate(): void
    {
        // TODO: implement escalate
    }

    public function resolve($resolution_text): void
    {
        // TODO: implement resolve
    }

    public function closeResolved(): void
    {
        // TODO: implement close_resolved
    }

    public function review(): void
    {
        // TODO: implement review
    }

}
